### PR TITLE
Docs Backlog Update

### LINKED ISSUE #141: Add 'creator' component documentation
for both json & markdown.

createRemarkButton --> remarkCreatorPlugin?

### LINKED ISSUE #142: Document how to create new JSON files
piggybacking off of #141 — currently we have no creating 'new' files with JSON documented.

### LINKED ISSUE #147: Update pathToRepo with better example
https://github.com/tinacms/tinacms/blob/master/packages/demo-gatsby/gatsby-config.js#L20

in the configure-fit-plugin doc

### PR DESCRIPTION
**Pull request by kendallstrautman**
Fixes #147 
Fixes #141 
Fixes #142 - although we need to add more examples of using the JsonCreator alongside the RemarkCreator

### CODE REVIEW

File: content/docs/gatsby/creating-new-files.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+      name: 'filename',
+      component: 'text',
+      label: 'Filename',
+      placeholder: 'content/blog/hello-world/index.md',
+      description: 'The full path to the new markdown file, relative to the repository root.',
+    },
+  ],
+})
+```
+<tip>
```
Comment (DirtyF) [inline content/docs/gatsby/creating-new-files.md]: Can we specify in what version the createRemarkButton was deprecated here?

File: content/docs/gatsby/creating-new-files.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+})
+```
+
+**JSON Example**
+
+```javascript
+import { JsonCreatorPlugin } from 'gatsby-tinacms-json'
+
+const CreatePostPlugin = new JsonCreatorPlugin({
+  label: 'Create New File',
```
Comment (DirtyF) [inline content/docs/gatsby/creating-new-files.md]:
```suggestion
  label: 'New JSON File',
```

File: content/docs/gatsby/creating-new-files.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+- `filename`: A function whose return value should be the path to the new file.
+- `fields`: An array of field objects. Read more on field defitions [here](/docs/concepts/fields).
+
+**Markdown Example**
+
+```javascript
+import { RemarkCreatorPlugin } from 'gatsby-tinacms-remark'
+
+const CreatePostPlugin = new RemarkCreatorPlugin({
+  label: 'Add New Post',
```
Comment (DirtyF) [inline content/docs/gatsby/creating-new-files.md]:
```suggestion
  label: 'New Blog Post',
```

File: content/docs/gatsby/creating-new-files.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
+---
+title: Create New Files
```
Comment (DirtyF) [inline content/docs/gatsby/creating-new-files.md]:
```suggestion
title: Creating New Files
```

File: content/docs/gatsby/creating-new-files.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+id: /docs/gatsby/create-new-files
```
Comment (DirtyF) [inline content/docs/gatsby/creating-new-files.md]:
```suggestion
id: /docs/gatsby/creating-new-files
```

File: content/docs/gatsby/creating-new-files.md (hunk 1 above)
Comment (DirtyF) [reply content/docs/gatsby/creating-new-files.md]: Also it's more a warning than a tip, no?

Comment (kendallstrautman) [reply content/docs/gatsby/creating-new-files.md]: lol true, we could make a warning component. I think for now the tip component works. I'll check out the changelogs to see if I can track down a version

**Pull request merged**